### DISCUSSION TITLE
The ability to delete Detected spam

### DISCUSSION THREAD

**Discussion by proarachnid** [umputun/tg-spam#76] [Ideas]
Dear developer, it would be great to delete detected spam at least through the web muzzle. Especially at the testing stage, a lot of records appear there and most of them quickly lose their relevance. If the trash icon appeared next to the record, it would be great!

**Comment by umputun** [comment]
The system already has the ability to remove spam via the web UI, as well as unban and remove spam samples from the admin chat. So, I'm not sure what's missing here.

UI Spam Samples:

<img width="663" alt="image" src="https://github.com/umputun/tg-spam/assets/535880/b1ed97c9-e999-4557-85da-71376e832ae5">


admin channel/group:

<img width="613" alt="image" src="https://github.com/umputun/tg-spam/assets/535880/b98298a8-47ed-4cb4-b9e3-748ba0e471fd">

**Reply by proarachnid** [reply]
We can actually delete any samples both in files and in the web interface. There is a wonderful basket icon for this. But I was talking about the Detected Spam tab. The data that is present there is in the database. Over time, the information in this tab loses its relevance, especially if we ourselves test the behavior of the bot for a long time and send it stop words and other samples ourselves. It would be convenient to delete this outdated and irrelevant information.
<img width="1931" alt="Снимок экрана 2024-04-29 в 12 35 05" src="https://github.com/umputun/tg-spam/assets/67074774/8f8e245e-788c-46c4-a148-c6ac868af147">

**Reply by umputun** [reply]
The tab represents spam detected automatically and indicates what conditions/checks triggered it. Removing this data will not change the behavior of the spam detector at all because all the tuning is done with spam and ham lists, and this is rather the end result. I'm not sure what "outdated and irrelevant information" even means in this context. This was a spam detection for real, and this detection was recorded. Practically, this data stored in the indexed SQLite table, and it needs to grow way too big (tens of millions of reports) to cause any practical concerns.

So, I don't see a good reason to implement this functionality with manual deletes. Alternatively, I can suggest a relatively simple change allowing to cap the size of this collection with an optional parameter defining how many latest records kept in the `detected_spam` table. If you want to offer a PR for this auto-cleanup, I'll accept it.

**Reply by proarachnid** [reply]
You are right, the existence of these records does not affect the actual operation of the bot in any way. My suggestion referred exclusively to "cosmetics". When I visit this page, I constantly see test responses for words and phrases that are no longer in the samples. That's why I proposed this idea in order to be able to see real combat triggers on this page, and not test words. I'm sorry, it's just my perfectionism. You're right. There will be no practical benefit from this.
